packages/opentelemetry-sdk-trace-base/src/BasicTracerProvider.ts

```diff
@@ -89,20 +88,20 @@ export class BasicTracerProvider implements TracerProvider {
       resource: this.resource,
     });
 
+    const spanProcessors: SpanProcessor[] = [];
+
     if (config.spanProcessors?.length) {
-      this._registeredSpanProcessors = [...config.spanProcessors];
-      this.activeSpanProcessor = new MultiSpanProcessor(
-        this._registeredSpanProcessors
-      );
+      spanProcessors.push(...config.spanProcessors);
     } else {
       const defaultExporter = this._buildExporterFromEnv();
-      if (defaultExporter !== undefined) {
-        const batchProcessor = new BatchSpanProcessor(defaultExporter);
-        this.activeSpanProcessor = batchProcessor;
-      } else {
-        this.activeSpanProcessor = new NoopSpanProcessor();
-      }
+      spanProcessors.push(
+        defaultExporter
+          ? new BatchSpanProcessor(defaultExporter)
+          : new NoopSpanProcessor()
+      );
     }
+
+    this.activeSpanProcessor = new MultiSpanProcessor(spanProcessors);
```

Comment:
note to reviewer: `forceFlush` method was only iterating over the `_registeredSpanProcessors`. So when using the default exporter and processor (the else case) the `BatchSpanProcessor.forceFlush` method was never called. I've added a test for it